### ISSUE TITLE
git Integration synchronizes both ways without an option to disable

### ISSUE THREAD

**Issue by jantari** [ironmansoftware/powershell-universal-legacy#45]
#### Description

According to the documentation: https://docs.ironmansoftware.com/config/git#git-synchronization-behavior

> If you setup git sync with a preexisting git remote, the changes will be pulled and synchronized locally. You cannot have any changes locally.

Which implies a one-way (pull) sync from the remote git repository. This is of course ideal, as git-platforms like GitHub and GitLab are often needed for code review, automatic linting, change management etc.

However, it turns out that PowerShell Universal 1.3.2 actually does synchronize both ways instead - it will commit and push on a regular interval, syncing all local changes back to the remote git repository. This of course completely bypasses any code review guidelines or change management practices, and the commit messages produced are not in line with any/our standard either:
![image](https://user-images.githubusercontent.com/25607767/93855407-bdb21200-fcb7-11ea-85a1-915caff137b2.png)

There needs to be an option to disable the two-way sync and exclusively do fetching/pulling from Universal Automation, so existing git-workflow processes are not bypassed.

#### Additional Information

Remote repository is a private GitHub repository that's connected to via HTTPS and a PAT.

#### Version Information

- Universal: 1.3.2
- Universal Hosting Method (IIS, Executable, docker etc): Executable (Kestrel) from ZIP
- PowerShell: 5.1 and 7.0.3
- Operating System: Windows Server 2019 (Build 10.0.17763.1457)

#### Internal Issue

We'll fill this out when we create an internal issue in our repository.

**Comment by adamdriscoll** [comment]
@jantari - I agree.

Can you please let me know what is included in those commits in your environment? We will have to make the Web API and Admin console read-only to avoid having to send changes from the local instance back to the remote and I'm curious what changes are being made locally to cause the git sync to happen like this.

**Comment by jantari** [comment]
In chronological order:

1. Created a new empty file `.universal/scripts.ps1`
2. Created a file `.universal/endpoints.ps1` with content:
```plain
New-PSUEndpoint -Url "/test-api" -Method "POST" -Endpoint { 
# Enter your script to process requests.
} 
```
3. Added content to `.universal/scripts.ps1`:
```plain
New-PSUScript -Name "scripts.ps1" -Description "This is a required configuration script for PowerShell Universal" -Path "scripts.ps1" 
```
Created file `scripts.ps1` with content:
```plain
# scripts.ps1
```
(this was maybe me through the WebUI)
4. Added ` -InformationAction "Continue"` to the end of the `New-PSUScript` command in `.universal/scripts.ps1`

Commits 5 and 6 was me editing scripts in the WebUI.

7. Added a file `.universal/variables.ps1` with content:
```plain
New-PSUVariable -Name "Sophos Central Login" -Vault "BuiltInLocalVault" -Type "PSCredential"
```

As for the ability to edit scripts in the WebUI, it should probably be greyed out/disabled
when a one-way git sync is configured. It would be rather confusing to have the ability to edit there,
but then have your changes reset/overwritten immediately by a git re-sync.

**Comment by adamdriscoll** [comment]
Agreed. We'd have to make the Web UI and API read-only if one way sync were enabled. 

Thanks for the info. This will likely fall into our 1.5 roadmap.

**Comment by jantari** [comment]
Hey, is there any update on the ETA for this? I follow the 1.5.x changelogs but so far no mention, if it's been postponed to 1.6.x or 1.7.x I don't have to monitor the changelogs all the time hat's why I ask

**Comment by adamdriscoll** [comment]
This just got merged into 1.5. It will be available in 1.5.13. You'll be able to set the git sync behavior in appsettings.json.

```
  "Data": {
    "RepositoryPath": "%ProgramData%\\UniversalAutomation\\Repository",
    "ConnectionString": "%ProgramData%\\UniversalAutomation\\database.db",
    "GitRemote": "https://github.com/ironmansoftware/pus-test.git",
    "GitUserName": "adamdriscoll",
    "GitPassword": "", 
    "GitBranch": "main",
    "GitSyncBehavior": "OneWay",
    "ConfigurationScript": ""
  },
```

It will disable Admin and Operator roles so the console is effectively readonly. If you have the Executor role, you will still be able to run jobs. It also disables any commits or pushes to git. 

We've added a banner to the top of the admin console so it's clear when this is enabled. 

![image](https://user-images.githubusercontent.com/1256531/107660033-0d807e00-6c45-11eb-926c-e2275f167544.png)

**Comment by adamdriscoll** [comment]
This was released this morning in 1.5.13: https://docs.ironmansoftware.com/config/git#one-way
